### DISCUSSION TITLE
Doesnt seem to work with Parsec

### DISCUSSION THREAD

**Discussion by HerronTsure** [complexlogic/flex-launcher#18] [General]
I have Parsec set up on a server. https://parsec.app/
I have flex-launcher set up on a client with Parsec.
I added an entry like this:
Entry1=Computer;.\assets\icons\computer.png;"C:\Program Files\Parsec\parsecd.exe" peer_id=PEER_ID

When I click in flex-launcher, Parsec launches, connects to the client, and goes full screen, as expected.
However, almost instantly flex-launcher pops back up to the top covering Parsec.
If I quit flex-launcher, Parsec is still running full screen connected to the server underneath.

**Comment by complexlogic** [comment]
It's possible that `parsecd.exe` is a forking executable, meaning that it forks a new process for the window and then immediately returns. Flex Launcher works by tracking the launched process handle. It is designed to work with blocking processes, i.e. the process doesn't return until the application has finished. When the process returns, the launcher assumes that the application has finished, and then will grab back the window focus. There was a similar issue reported [here](https://github.com/complexlogic/flex-launcher/issues/2).

When you launch Parsec, is the process image name still `parsecd.exe` or something else? You can check with Task Manager with the ctrl + shift + Esc keyboard shortcut.

**Reply by HerronTsure** [reply]
The app running in task manager is parsecd.exe
Does flex-launcher log the process ID anywhere? Then I could check if it was the same process that was launched.

**Reply by HerronTsure** [reply]
I see the issue. After the first time you start Parsec it runs in the system tray, so when you close the app it does not actually quit out.
When you call it the second time it does not launch a new app, just sends the command line to the already running one.

**Reply by complexlogic** [reply]
Steam behaves in the same way. So when you launch Parsec via Flex Launcher from a clean boot with Parsec not running yet, does it work in that case?

**Reply by HerronTsure** [reply]
Yes. If Parsec isn't running, it launches correctly. I am searching to see if there is a way to get Parsec to quit when you disconnect.
